test/conftest.py

```diff
@@ -78,6 +79,35 @@ def _run(cmd_string):
     return _run
 
 
+@pytest.fixture(scope="function")
+def daemon_runner(pytestconfig, data_dir, downloads_dir, working_dir):
+    """
+    Provide an invoke's `Local` object that has started the arduino-cli in daemon mode.
+    This way is simple to start and kill the daemon when the test is finished
+    via the kill() function
+
+    Useful reference:
+        http://docs.pyinvoke.org/en/1.4/api/runners.html#invoke.runners.Local
+        http://docs.pyinvoke.org/en/1.4/api/runners.html
+    """
+    cli_full_line = os.path.join(str(pytestconfig.rootdir), "..", "arduino-cli daemon")
+    env = {
+        "ARDUINO_DATA_DIR": data_dir,
+        "ARDUINO_DOWNLOADS_DIR": downloads_dir,
+        "ARDUINO_SKETCHBOOK_DIR": data_dir,
+    }
+    os.makedirs(os.path.join(data_dir, "packages"))
+    run_context = Context()
+    run_context.cd(working_dir)
+    # Local Class is the implementation of a Runner abstract class
+    runner = Local(run_context)
+    runner.run(
+        cli_full_line, echo=False, hide=True, warn=True, env=env, asynchronous=True
+    )
+
+    return runner
```

Comment:
If you `yield` instead of `return`, you can tear down the process in this function instead of having the function tests doing so. The pattern would be:
```
def my_fixture():
  ...
  runner.run(...)
  yield runner  # we block here until the test function using this fixture has returned
  kill(runner)
```